### PR TITLE
Add setting for enableing/disabling cloud ICE servers

### PR DESCRIPTION
**Pull request by klejejs**
## Proposed change

This PR adds support for enabling/disabling usage of the cloud ICE servers in the cloud settings page.
The feature itself is implemented in https://github.com/home-assistant/core/pull/128942.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/cloud/account/cloud-ice-servers-pref.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    return html`
+      <ha-card
+        outlined
+        header=${this.hass.localize(
+          "ui.panel.config.cloud.account.ice_servers.title"
+        )}
+      >
+        <div class="header-actions">
+          <a
+            href="https://www.nabucasa.com/config/"
```
Comment (klejejs) [inline src/panels/config/cloud/account/cloud-ice-servers-pref.ts:33]: This is yet to change.

File: src/translations/en.json [hunk 2]
```diff
@@ -3922,6 +3922,11 @@
               "certificate_expire": "Certificate renewal at {date}",
               "more_info": "More details"
             },
+            "ice_servers": {
+              "title": "Cloud WebRTC ICE servers",
+              "info": "Do you want to use Home Assistant Cloud for WebRTC connections? Enable this option to use the Home Assistant Cloud WebRTC ICE servers.",
+              "link_learn_how_it_works": "Learn how it works"
+            },
```
Comment (klejejs) [inline src/translations/en.json:3929]: These are yet to change as well.

**Comment by MartinHjelmare** [comment]
Core PR is merged.

File: src/translations/en.json (hunk 2 above)
Comment (bramkragten) [reply src/translations/en.json:3936]: When will that be ready?

File: src/panels/config/cloud/account/cloud-ice-servers-pref.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @property({ attribute: false }) public cloudStatus?: CloudStatusLoggedIn;
+
+  protected render() {
+    if (!this.cloudStatus) {
+      return nothing;
+    }
+
+    const { cloud_ice_servers_enabled } = this.cloudStatus.prefs;
```
Comment (bramkragten) [inline src/panels/config/cloud/account/cloud-ice-servers-pref.ts:22]: Please use camelcase for variables

File: src/translations/en.json (hunk 2 above)
Comment (klejejs) [reply src/translations/en.json]: This is now changed to the correct texts.

File: src/panels/config/cloud/account/cloud-ice-servers-pref.ts [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+  private async _toggleCloudICEServersEnabledChanged(ev) {
+    const toggle = ev.target as HaSwitch;
+
+    try {
+      await updateCloudPref(this.hass, {
+        cloud_ice_servers_enabled: toggle.checked,
+      });
+      fireEvent(this, "ha-refresh-cloud-status");
+    } catch (err: any) {
+      alert(err.message);
```
Comment (silamon) [inline src/panels/config/cloud/account/cloud-ice-servers-pref.ts:72]: We don't typically use these kind of alerts but handle it properly in the UI.

**Review by bramkragten — DISMISSED** [review]

**Review by piitaya — APPROVED** [review]

**Pull request merged**